docs/development.rst

```diff
@@ -33,11 +33,13 @@ Quickstart
 - Build the low level module by running ``make`` in the ``python`` directory.
 - Run the tests to ensure everything has worked: ``python -m nose -vs``. These should
   all pass.
-
-- Make your changes in a local branch, and open a pull request on GitHub when you
-  are ready. Please make sure that (a) the tests pass before you open the PR; and
-  (b) your code passes PEP8 checks (see :ref:`sec_development_code_style` below
-  for a git commit hook to ensure this happens automatically) before opening the PR.
+- Install the pre-commit checks: ``pre-commit install``
+- Make your changes in a local branch, on each commit checks will be run for code style
+  and common problems (see :ref:`sec_development_code_style`), sometimes these will
```

Comment:
This should be two sentences I think?

Reply:
also a bit more explanation would be helpful for people unfamiliar - e.g. say that the code changes are formatting only; that the code is guaranteed to do the same thing.

Reply:
Agreed, I've also added quite a bit more text about how to run checks without committing, and committing without checks.